preconditions-assertj/src/main/java/com/palantir/logsafe/testing/LoggableExceptionAssert.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * (c) Copyright 2018 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.logsafe.testing;
+
+import com.palantir.logsafe.Arg;
+import com.palantir.logsafe.SafeLoggable;
+import java.util.List;
+import org.assertj.core.api.AbstractThrowableAssert;
+import org.assertj.core.api.ListAssert;
+
+public class LoggableExceptionAssert<T extends Throwable & SafeLoggable>
+        extends AbstractThrowableAssert<LoggableExceptionAssert<T>, T> {
+    private final ArgsAssert argsAssert;
+
+    public LoggableExceptionAssert(T actual) {
+        super(actual, LoggableExceptionAssert.class);
+
+        if (actual == null) {
+            argsAssert = null;
+        } else {
+            argsAssert = new ArgsAssert(actual.getArgs());
+        }
+    }
+
+    public final LoggableExceptionAssert<T> hasExactlyArgs(Arg<?>... args) {
+        isNotNull();
+
+        argsAssert.containsExactlyInAnyOrder(args);
+        return this;
+    }
+
+    public final LoggableExceptionAssert<T> hasArgs(Arg<?>... args) {

```

Comment:
Not sure it's worth the distinction between `hasArgs` and `hasExactlyArgs` - at the end of the day I don't think these args should be considered to be ordered, so can we just condense to one method?

Reply:
`hasArgs` is it has as least those args, `hasExactArgs` is it contains only those args.  I think the distinction is worth it, but clearly the naming needs to be better.